mesa/agent.py

```diff
@@ -348,29 +348,103 @@ def agg(self, attribute: str, func: Callable) -> Any:
         values = self.get(attribute)
         return func(values)
 
-    def get(self, attr_names: str | list[str]) -> list[Any]:
+    def get(
+        self,
+        attr_names: str | list[str],
+        handle_missing: str = "error",
+        default_value: Any = None,
+    ) -> list[Any] | list[list[Any]]:
         """
         Retrieve the specified attribute(s) from each agent in the AgentSet.
 
         Args:
             attr_names (str | list[str]): The name(s) of the attribute(s) to retrieve from each agent.
+            handle_missing (str, optional): How to handle missing attributes. Can be
+
+                                            -'error' (default), raises an AttributeError if attribute is missing.
+                                            -'skip' which skips the agents missing the attribute.
+                                            'default' to return the specified `default_value`.
+
+            default_value (Any, optional): Used in conjunction with ``handle_missing='default``
+                                     The default value to return if 'handle_missing' is set to 'default'
+                                     and the agent does not have the attribute.
 
         Returns:
-            list[Any]: A list with the attribute value for each agent in the set if attr_names is a str
-            list[list[Any]]: A list with a list of attribute values for each agent in the set if attr_names is a list of str
+            list[Any]: A list of attribute values for each agent if attr_names is a str.
+            list[list[Any]]: A list of lists of attribute values for each agent if attr_names is a list of str.
```

Comment:
Minor point that can be ignored: if this were `dict[str, list[Any]]` where the keys are attribute names, the output could be readily turned into a DF.